integration-tests/contract-permits.spec.ts

```diff
@@ -0,0 +1,242 @@
+        .then((c) => {
+          return c.methods.permit(permit_parameter).send();
+        })
+        .then(async (op) => {
+            await op.confirmation(3);
+            return op.hash;
+          })
+          .catch(Error) ;
+          expect(op.hash).toBeDefined();
+          expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+
+        //submit 42 to the wrapped entrypoint
+        Tezos.contract
+        .at(contract.address)
+        .then((c) => {
+          return c.methods.wrapped("42").send();
+        })
+        .then(async (op) => {
+            await op.confirmation(3);
+            return op.hash;
+          })
+        .catch(Error) ;
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        done();
+      });
+
+  test('Deploy a contract having a permit with expiry', async (done) => {
+
+    const op = await Tezos.contract.originate({
+      code: permit_admin_42_expiry,
+      storage:
+        {
+        0: 300,
+        1: new MichelsonMap(),
+        2: 0,
+        3: 'tz1h1LzP7U8bNNhow8Mt1TNMxb91AjG3p6KH',
+        4: 'tz1h1LzP7U8bNNhow8Mt1TNMxb91AjG3p6KH'
+        },
+      });
+      await op.confirmation();
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+      const contract = await op.contract();
+      expect(op.status).toEqual('applied')
+
+      const expiry = 300;
+
+      //submit the expiry to the contract
+       Tezos.contract
+       .at(contract.address)
+       .then((c) => {
+         return c.methods.setExpiry(expiry).send();
+       })
+       .then(async (op) => {
+         await op.confirmation(3);
+         return op.hash;
+       })
+       .catch(Error) ;
+       expect(op.hash).toBeDefined();
+       expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+       done();
+    });
+
+    test('Deploy a contract having a permit that can be set', async (done) => {
+      const op = await Tezos.contract.originate({
+        code: permit_admin_42_set,
+        storage:
+        {
+        0: new MichelsonMap(),
+        1: 300,
+        2: 'tz1h1LzP7U8bNNhow8Mt1TNMxb91AjG3p6KH',
+        3: 'tz1h1LzP7U8bNNhow8Mt1TNMxb91AjG3p6KH'
+        },
+        });
+        await op.confirmation();
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const contract = await op.contract();
+        expect(op.status).toEqual('applied')
+
+        //anyone can submit 42 to the contract in place of bob
+        Tezos.contract
+        .at(contract.address)
+        .then((c) => {
+          return c.methods.permit("42").send();
+        })
+        .then(async (op) => {
+          await op.confirmation(3);
+          return op.hash;
+        })
+        .catch(Error) ;
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
```

Comment:
The lines 235 and 236 are testing again the constant op created at line 201 instead of the operation representing the call of the permit method

Reply:
I removed the lines